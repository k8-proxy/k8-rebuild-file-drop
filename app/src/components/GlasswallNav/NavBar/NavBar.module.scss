.navBar {
    position: relative;
    float: left;
    top: 0;
    width: 7rem;
    height: 100%;
    letter-spacing: .2rem;
    background: #1b5c7e;
    background: linear-gradient(to top, #196480, #0c3450);
    font-family: "robotoregular", sans-serif;
    overflow: hidden;
    transition: all .3s ease-in-out;

    &.expanded {
        width: 24rem;
    }

}

.loading {
    color: #fff;
    text-align: center;
    padding-top: 4rem;
    font-size: 1.4rem;
}

.logo {
    width: 100%;
    height: 8rem;
    margin: 3rem 0;
}