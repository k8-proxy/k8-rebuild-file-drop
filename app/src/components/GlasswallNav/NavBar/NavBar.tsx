import React from "react";
import PropTypes from "prop-types";

import GlasswallLogo from "../../GlasswallLogo/GlasswallLogo";

import styles from "./NavBar.module.scss";

export interface NavBarProps { logo: boolean, expanded: boolean, children: React.ReactNode }

const NavBar = (props: NavBarProps) => {

    return (
        <div className={`${styles.navBar} ${props.expanded ? styles.expanded : ""}`}>
            {props.logo &&
                <GlasswallLogo className={styles.logo} />
            }

            {props.children}

        </div>
    );
};

NavBar.propTypes = {
    logo: PropTypes.bool
};

export { NavBar };