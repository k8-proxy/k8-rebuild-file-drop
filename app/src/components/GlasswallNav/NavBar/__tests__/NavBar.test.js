import React from "react";
import { mount } from "enzyme";
import TestRenderer from "react-test-renderer";
import { NavBar } from "../NavBar";

const logo = false;
const expanded = true;

const props = {
    logo: logo,
    expanded: expanded
};

test("NavBar_Snapshot", () => {
    // Arrange
    const component = TestRenderer.create(<NavBar {...props} />);

    // Act
    let tree = component.toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
});

test("Displays_Correct_Props", () => {
    // Arrange
    // Act
    const navBarComponent = mount(<NavBar {...props}/>);

    // Assert
    expect(navBarComponent.prop("logo")).toEqual(props.logo);
    expect(navBarComponent.prop("expanded")).toEqual(props.expanded);
});