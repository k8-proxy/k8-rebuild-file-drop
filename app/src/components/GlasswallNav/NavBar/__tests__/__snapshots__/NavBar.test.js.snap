// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar_Snapshot 1`] = `
<div
  className="navBar expanded"
/>
`;
